app/assets/javascripts/discourse/app/controllers/composer.js

```diff
@@ -402,18 +402,35 @@ export default Controller.extend({
   //
   // opts:
   //
-  // - fallbackToNewTopic: if true, and there is no draft, the composer will
-  // be opened with the create_topic action and a new topic draft key
+  // - topic: if this is present, the composer will be opened with the reply
+  // action and the current topic key and draft sequence
+  // - fallbackToNewTopic: if true, and there is no draft and no topic,
+  // the composer will be opened with the create_topic action and a new
+  // topic draft key
   // - insertText: the text to append to the composer once it is opened
   // - openOpts: this object will be passed to this.open if fallbackToNewTopic is
-  // true
+  // true or topic is provided
   @action
   focusComposer(opts = {}) {
     if (this.get("model.viewOpen")) {
       this._focusAndInsertText(opts.insertText);
     } else {
       const opened = this.openIfDraft();
-      if (!opened && opts.fallbackToNewTopic) {
+      if (!opened && opts.topic) {
+        this.open(
+          Object.assign(
+            {
+              action: Composer.REPLY,
+              draftKey: opts.topic.get("draft_key"),
+              draftSequence: opts.topic.get("draft_sequence"),
+              topic: opts.topic,
+            },
+            opts.openOpts || {}
+          )
+        ).then(() => {
+          this._focusAndInsertText(opts.insertText);
+        });
+      } else if (!opened && opts.fallbackToNewTopic) {
```

Comment:
I am thinking if we can make all that nested ifs a little bit easier to follow. This function is doing 2 things: open composer and insert text. Some if branches are returning promise and some not so `linkLookup` must be repeated few time.

What if we would extract another function like:

```javascript
openComposer(opts) {
  if (this.get("model.viewOpen")) {
    return Promise.resolve()
  } else {
    const opened = this.openIfDraft();
    if(opened) {
      return Promise.resolve()
    }
    if (opts.topic) {
      this.open(
        Object.assign(
          {
            action: Composer.REPLY,
            draftKey: opts.topic.get("draft_key"),
            draftSequence: opts.topic.get("draft_sequence"),
            topic: opts.topic,
          },
          opts.openOpts || {}
        )
....
  }
}
```

In that case final function can be
```javascript
focusComposer(opts = {}) {
  this.openComposer(opts).then(() => {
    this._focusAndInsertText(opts.insertText);
  });
}
```

Reply:
Ah yes that is nice, will change that!